src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Item.js

```diff
@@ -1,28 +1,84 @@
 // @flow
-import Icon from '../../components/Icon';
-import type {Item as ItemType} from './types';
+import {action, observable} from 'mobx';
 import React from 'react';
+import classNames from 'classnames';
+import {observer} from 'mobx-react';
+import type {Item as ItemType} from './types';
+import Icon from '../../components/Icon';
 import itemStyles from './item.scss';
+import Dropdown from './Dropdown';
 
+@observer
 export default class Item extends React.PureComponent {
     props: ItemType;
 
     static defaultProps = {
         enabled: true,
     };
 
+    @observable hasDropdownOpen = false;
+
+    @action toggleDropdown = () => {
+        this.hasDropdownOpen = !this.hasDropdownOpen;
+    };
+
+    @action closeDropdown = () => {
+        if (this.hasSubItems) {
+            this.hasDropdownOpen = false;
+        }
+    };
+
+    componentWillMount() {
+        const {items} = this.props;
+
+        this.hasSubItems = items && items.length;
```

Comment:
Why not building that as an own method? Then we wouldn't have to take care of another variable on our own.